erpnext/selling/doctype/sales_order/sales_order.py

```diff
@@ -1071,3 +1071,210 @@ def update_produced_qty_in_so_item(sales_order, sales_order_item):
 	if not total_produced_qty and frappe.flags.in_patch: return
 
 	frappe.db.set_value('Sales Order Item', sales_order_item, 'produced_qty', total_produced_qty)
+
+@frappe.whitelist()
+def make_production_plan(source_name, target_doc=None):
+	def set_missing_values(source, target):
+		target.get_items_from = source.doctype
+		target.append("sales_orders", {
+			"sales_order": source.name,
+			"sales_order_date": source.transaction_date,
+			"customer": source.customer,
+			"grand_total": source.base_grand_total
+		})
+
+	# Updates the fields of items.
+	def update_item_data(source, target, source_parent):
+		bom_no = frappe.get_value("BOM", {
+			"item": target.item_code,
+			"is_default": 1
+			})
+		if bom_no:
+			bom_data = frappe.db.get_value('BOM', bom_no, ['raw_material_cost','operating_cost'], as_dict=1)
+			target.bom_no = bom_no
+			target.raw_material_cost = bom_data.raw_material_cost
+			target.total_operational_cost = bom_data.total_operating_cost
+			target.total_operational_hours = flt(frappe.db.get_value("BOM Operation", {"parent": bom_no}, "sum(time_in_mins)")) / 60.0
+			target.total_workstations = frappe.db.count("BOM Operation", {"parent": bom_no, "workstation": ["!=", ""]})
+
+	# Mapping Sales Order doc to new Production Plan doc.
+	doc = get_mapped_doc("Sales Order", source_name, {
+		"Sales Order": {
+			"doctype": "Production Plan",
+		},
+		"Sales Order Item": {
+			"doctype": "Production Plan Item",
+			"field_map": {
+				"item_code": "item_code",
+				'name': 'sales_order_item',
+				"qty": "planned_qty",
+				"parent": "sales_order"
+			},
+			'postprocess': update_item_data
+		}
+	}, target_doc, set_missing_values)
+	return doc
+
+@frappe.whitelist()
+def create_multiple_pick_lists(orders):
+	"""Creating different Pick Lists from multiple Sales Order."""
+	orders = json.loads(orders)
+
+	created_orders = []
+	for order in orders:
+		created = False
+		customer = frappe.db.get_value("Sales Order", order, "customer")
+
+		# check if a Pick List already exists against the order.
+		pick_lists = frappe.get_all("Pick List",
+			filters=[
+				["Pick List", "docstatus", "<", 2],
+				["Pick List Item", "sales_order", "=", order]
+			],
+			distinct=True)
+		pick_lists = [item.name for item in pick_lists if item.name]
```

Comment:
```suggestion
```
You dont have to loop over it again. ORM will always return data. Use `as_list` param for get_all